wallet/src/de/schildbach/wallet/ui/CreateUsernameActivity.kt

```diff
@@ -78,44 +85,100 @@ class CreateUsernameActivity : InteractionAwareActivity(), TextWatcher {
             showCompleteState()
         }
 
+        walletApplication = application as WalletApplication
+
         val confirmTransactionSharedViewModel = ViewModelProviders.of(this)
                 .get(SingleActionSharedViewModel::class.java)
         confirmTransactionSharedViewModel.clickConfirmButtonEvent.observe(this, Observer {
+            // setup the wallet
+            val wallet = (application as WalletApplication).wallet
```

Comment:
We can use `walletApplication.wallet`, since `walletApplication` was already defined in line 88.

Reply:
excellent

Reply:
I think that was changed here:  https://github.com/dashevo/dash-wallet/pull/393/commits/07f06f10cd749c36d273323945784ef61d47cbae